sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -194,6 +203,21 @@ public void SetProperty(string name, object value)
             }
         }
 
+        /// <summary>
+        /// Returns the value of the <see cref="Response"/> property and
+        /// transfers dispose ownership of the response to the caller. After
+        /// calling this method, the <see cref="Response"/> property will be
+        /// null and the caller will be responsible for disposing the returned
+        /// Response value, which may hold a live network stream.
+        /// </summary>
+        public Response TransferResponseDisposeOwnership()
```

Comment:
nit: should we consider a name that is more explicit about where the ownership is being transferred?
```suggestion
        public Response AssumeResponseDisposeOwnership()
```

Reply:
I felt that using the word "transfer" was more consistent with the terminology used by Roslyn, which calls this a ["dispose ownership transfer"](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2000).  Since we return the response from the method, it seems fairly clear to me that the person who called the method and is holding the returned value is the person who the ownership has been transferred to.  But I'm happy to defer to architects regarding what they think is optimal naming for the final API -- @KrzysztofCwalina, @tg-msft?

Reply:
I don't have a name that I feel strongly about. I don't think any name will be completely self explanatory. Having said that, I thought about something like ExtractResponse or DetatchResponse. ... We can also talk with the BCL team. Maybe they have good suggestions.

Reply:
We use Extract in ExtractResponseContent. I think we should stick to it.

Reply:
Sounds good.  I will scope the change to ClientModel for now, and we can start a thread with the BCL team about the name of this method addition to PipelineMessage.